frontend/src/redux/Account.ts

```diff
@@ -0,0 +1,31 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+/**
+ * Due to serialization issues, we can't directly store the
+ * firebase.User object in Redux (instead we store a toJson
+ * object version). This means we'll have to manually specify
+ * the type and copy any relevant fields from the docs below:
+ * https://firebase.google.com/docs/reference/js/firebase.User
+ */
+export type AccountType = {
+  displayName: string,
+  email: string,
+  emailVerified: boolean,
+  refreshToken: string,
+  uid: string,
```

Comment:
Why are these attributes needed on the frontend? I could potentially see users for `uid` as the `userId` as we've briefly discussed, but not sure about these attributes' uses in this PR?

Reply:
These come from Firebase's [User type](https://firebase.google.com/docs/reference/js/firebase.User), which includes these properties plus a lot of other miscellaneous fields. Let me know if you want me to remove all but the uid and email fields - the other fields I think will still exist in the actual objects, but if we remove them from the TypeScript definition then we'd basically be able to ignore them/act like they don't exist

Reply:
If we don't use them here, I think it would make sense to remove them, similar to how we only include relevant fields in DTOs. If for some reason you want to keep all of them though, that's OK.

Reply:
Sounds good - I removed everything but uid and email, which are two fields that we do use.